kod/object/passive/spell/umbrella.kod

```diff
@@ -125,6 +127,25 @@ messages:
       propagate;
    }
 
+   BreakTrance(who = $, state = $)
+   "Clears trance, removes enchantment(s), and sends notifications to any previously enchanted players"
+   {
+      local oUser,lEnchanted,iSpellpower;
+
+      Send(who,@ClearTranceFlag);
+      
+      iSpellpower = Nth(state,3);
+      lEnchanted = Nth(state,2);
+      
+      for oUser in lEnchanted
+      {
+         Send(oUser,@MsgSendUser,#message_rsc=Umbrella_break_trance);
+         Send(oUser,@RemoveEnchantment,#what=self);
```

Comment:
The original spell called `RemoveEnchantment` on the caster and then again on all enchanted, which in my testing included the caster and would throw a debug message. `RemoveEnchantment` has the same effect when you dig into it.

The original `BreakTrance`:
```
BreakTrance( who = $, state = $ )
{
      local oUser,lEnchanted,iSpellpower;

      iSpellpower = Nth(state,3);

      % Clean up Trance.
      Send(who,@ClearTranceFlag);
      Send(who,@RemoveEnchantment,#what=self);

      lEnchanted = Nth(state,2);
      for oUser in lEnchanted
      {
         Send(oUser,@MsgSendUser,#message_rsc=Umbrella_break_trance);
         Send(self,@LeaveUmbrella,#who=oUser,#iSpellpower=iSpellpower);
      }

      return;
}
```